### DISCUSSION TITLE
Can't get Vulnerability Detector to work

### DISCUSSION THREAD

**Discussion by giterest** [wazuh/wazuh#20186] [Q&A]
Hi,
I'm on Wazuh 4.6.0, agents have been upgraded to the latest version as well.

The out-of-the-box functionality to retrieve the feeds doesn't work, so I need to manually wget' them from the same machine (this works fine, so I came up with a script that downloads the feeds). Additionally, I changed to <os path> directives (for offline updates).

**Server, /var/ossec/etc/ossec.conf (Sorry, for whatever reason, code-formatting doesn't work);**
  <vulnerability-detector>
    
    <enabled>yes</enabled>
    <interval>5m</interval>
    <min_full_scan_interval>6h</min_full_scan_interval>
    <run_on_start>yes</run_on_start>

    
    <provider name="canonical">
      <enabled>yes</enabled>
      <os path="/usr/local/bin/wazuh-vul-feeds/com.ubuntu.jammy.cve.oval.xml.bz2">jammy</os>
      <update_interval>1h</update_interval>
    </provider>

    
    <provider name="debian">
      <enabled>yes</enabled>
      <os path="/usr/local/bin/wazuh-vul-feeds/oval-definitions-bullseye.xml.bz2">bullseye</os>
      <os path="/usr/local/bin/wazuh-vul-feeds/oval-definitions-bookworm.xml.bz2">bookworm</os>
      <path>/usr/local/bin/wazuh-vul-feeds/debian_security_tracker.json</path>
      <update_interval>1h</update_interval>
    </provider>
  </vulnerability-detector>

**Satellite, /var/ossec/etc/ossec.conf (Sorry, for whatever reason, code-formatting doesn't work);**
  
  <wodle name="syscollector">
    <disabled>no</disabled>
    <interval>1h</interval>
    <scan_on_start>yes</scan_on_start>
    <hardware>yes</hardware>
    <os>yes</os>
    <network>yes</network>
    <packages>yes</packages>
    <ports all="no">yes</ports>
    <hotfixes>yes</hotfixes>
    <processes>yes</processes>

    
    <synchronization>
      <max_eps>10</max_eps>
    </synchronization>
  </wodle>

This is from the logs, right after a restart of wazuh-manager on the server and wazuh-agent so you see what happens (filtered for vul;
`2023-11-11 @ 12:27:25.000 wazuh-modulesd:vulnerability-detector INFO  (5430): The update of the 'Debian Bookworm' feed finished successfully.
2023-11-11 @ 12:27:25.000 wazuh-modulesd:vulnerability-detector INFO  (5431): Starting vulnerability scan.
2023-11-11 @ 12:27:25.000 wazuh-modulesd:vulnerability-detector ERROR  (5582): Unavailable vulnerabilities at the NVD database. The scan is aborted.
2023-11-11 @ 12:27:19.000 wazuh-modulesd:vulnerability-detector INFO  (5430): The update of the 'Debian Bullseye' feed finished successfully.
2023-11-11 @ 12:27:19.000 wazuh-modulesd:vulnerability-detector INFO  (5400): Starting 'Debian Bookworm' database update.
2023-11-11 @ 12:27:12.000 wazuh-modulesd:vulnerability-detector INFO  (5430): The update of the 'Ubuntu Jammy' feed finished successfully.
2023-11-11 @ 12:27:12.000 wazuh-modulesd:vulnerability-detector INFO  (5400): Starting 'Debian Bullseye' database update.
2023-11-11 @ 12:26:59.000 wazuh-modulesd:vulnerability-detector INFO  (5400): Starting 'Ubuntu Jammy' database update.`

**So it comes down to that..**
`wazuh-modulesd:vulnerability-detector ERROR  (5582): Unavailable vulnerabilities at the NVD database. The scan is aborted.`
..and I don't know how to solve it.

That's the content of my feeds directory;
-rw-r--r-- 1 wazuh wazuh  2309961 Nov 11 07:57 com.ubuntu.jammy.cve.oval.xml.bz2
-rw-r--r-- 1 wazuh wazuh 35110437 Nov 10 14:51 debian_security_tracker.json
-rw-r--r-- 1 wazuh wazuh  2971697 Nov 10 11:39 oval-definitions-bookworm.xml.bz2
-rw-r--r-- 1 wazuh wazuh  3020488 Nov 10 11:38 oval-definitions-bullseye.xml.bz2

**Comment by pereyra-m** [comment]
Hello @giterest !

> The out-of-the-box functionality to retrieve the feeds doesn't work

Can you provide the `ossec.log` file of the server? There we could find more information about the feed download issue.
But if you are going to set the offline feed, there is a documentation section about it: https://documentation.wazuh.com/current/user-manual/capabilities/vulnerability-detection/offline-update.html

This error message
```
wazuh-modulesd:vulnerability-detector ERROR  (5582): Unavailable vulnerabilities at the NVD database. The scan is aborted.
```

means that you need to set up the offline download for the NVD because this feed is required for all the OSs available.

Regards.

**Comment by Dwordcito** [comment]
Hello, thank you for being part of the community, we closed this discussion due to inactivity, if it returns please reopen it and we will respond as soon as possible.
